File: rb/data_structures/binary_tree.rb
class Node 
  attr_accessor :value, :left, :right
  def initialize(value=nil, left=nil, right=nil)
    @value = value
    @left, @right = left, right
  end
end

class Tree
  attr_accessor :root
  def initialize()
    @root = nil
  end

  def to_object
    return @root
  end

  def add(value)
    curr = @root
    if curr == nil
      @root = Node.new(value, nil, nil)
    end
    while curr != nil
        if (value < curr.value) && (curr.left == nil)
            curr.left = Node.new(value,nil,nil)
        elsif  (value > curr.value) && (curr.right == nil)
            curr.right = Node.new(value,nil,nil)
        elsif (value < curr.value)
            curr = curr.left
        elsif (value > curr.value)
            curr = curr.right
        else
            return
        end
    end
  end
end

File: py/data_structures/binary_tree.py
class Node(object):
  """Node class for use in binary tree"""
  def __init__(self, value, left=None, right=None):
    super(Node, self).__init__()
    self.value = value
    self.left = left
    self.right = right

class Tree(object):
  """Tree class for use in binary tree"""
  def __init__(self):
    super(Tree, self).__init__()
    self.root = None

  def add(self, value):
    curr = self.root
    if curr == None:
      self.root = Node(value, None, None)
    while curr != None:
      if (value < curr.value and curr.left == None):
        curr.left = Node(value, None, None)
      elif value > curr.value and curr.right == None:
        curr.right = Node(value, None, None)
      elif value < curr.value:
        curr = curr.left
      elif value > curr.value:
        curr = curr.right
      else:
        return
